### PR TITLE
token-cli: reverse logic for --recipient-is-ata-owner

### PR DESCRIPTION
**Pull request by SUMEETRM**
Pull request for issue https://github.com/solana-labs/solana-program-library/issues/1805

Retained --recipient-is-ata-owner as a hidden flag and, if passed, warn the user that the behavior is now default, the flag is deprecated and can be removed.

New flag called --no-recipient-is-ata-owner to match convention of other CLI tools. Reverse the logic for the flag.

### CODE REVIEW

File: token/cli/src/main.rs [hunk 1]
```diff
@@ -2086,8 +2086,8 @@ fn app<'a, 'b>(
                         .help("Send tokens to the recipient even if the recipient is not a wallet owned by System Program."),
                 )
                 .arg(
-                    Arg::with_name("recipient_is_ata_owner")
-                        .long("recipient-is-ata-owner")
+                    Arg::with_name("no_recipient_is_ata_owner")
+                        .long("no-recipient-is-ata-owner")
                         .takes_value(false)
                         .requires("sign_only")
                         .help("In sign-only mode, specifies that the recipient is the owner of the associated token account rather than an actual token account"),
```
[the remark this answers is not in the document]
Comment (SUMEETRM) [reply token/cli/src/main.rs]: Fixed in the most recent commit

File: token/cli/src/main.rs [hunk 2]
```diff
@@ -2085,13 +2091,22 @@ fn app<'a, 'b>(
...
+                        .long("no-recipient-is-ata-owner")
+                        .takes_value(false)
+                        .requires("sign_only")
+                        .help("In sign-only mode, specifies that the recipient is the owner of the associated token account rather than an actual token account"),
+                )
                 .arg(
                     Arg::with_name("recipient_is_ata_owner")
                         .long("recipient-is-ata-owner")
                         .takes_value(false)
+                        .hidden(true)
```
Comment (joncinque) [inline token/cli/src/main.rs:2105]: Please also add a `conflicts_with("no_recipient_is_ata_owner")` so only one can be specified

File: token/cli/src/main.rs [hunk 3]
```diff
@@ -2831,10 +2846,10 @@ fn process_command(
             let allow_unfunded_recipient = arg_matches.is_present("allow_empty_recipient")
                 || arg_matches.is_present("allow_unfunded_recipient");
 
-            let recipient_is_ata_owner = arg_matches.is_present("recipient_is_ata_owner");
+            let no_recipient_is_ata_owner = arg_matches.is_present("no_recipient_is_ata_owner");
             let use_unchecked_instruction = arg_matches.is_present("use_unchecked_instruction");
             let memo = value_t!(arg_matches, "memo", String).ok();
-
+            let recipient_is_ata_owner = arg_matches.is_present("recipient_is_ata_owner");
```
Comment (joncinque) [inline token/cli/src/main.rs:2852]: Let's remove the `recipient_is_ata_owner` variable here, and also from `command_transfer` and instead have:
```
let no_recipient_is_ata_owner = arg_matches.is_present("no_recipient_is_ata_owner") || !arg_matches.is_present("recipient_is_ata_owner");
if arg_matches.is_present("recipient_is_ata_owner") {
    println_display(config, format!("recipient-is-ata-owner is now the default behavior. The option has been deprecated and will be removed in a future release."));
}
```

**Review by joncinque — COMMENTED** [review]
Almost there! Don't forget to run `cargo fmt` and to rebase this branch onto master

**Comment by joncinque** [comment]
Looking good! Now you just need to rebase onto master, fix the merge conflict, then we can get this in

**Pull request closed without merging**